internal/cmd/templates.go

```diff
@@ -0,0 +1,381 @@
+		Run: func(runner exec.Runner, args []string, flags *flags) error {
+			return runner.ListLintGroup(args[0])
+		},
+	}
+
+	serviceDescriptorProtoCmdTemplate = &cmdTemplate{
+		Use:   "service-descriptor-proto dirOrProtoFiles... servicePath",
+		Short: "Get the service descriptor proto for the service path.",
+		Args:  cobra.MinimumNArgs(1),
+		Run: func(runner exec.Runner, args []string, flags *flags) error {
+			return runner.ServiceDescriptorProto(args)
+		},
+		BindFlags: func(flagSet *pflag.FlagSet, flags *flags) {
+			flags.bindDirMode(flagSet)
+		},
+	}
+
+	versionCmdTemplate = &cmdTemplate{
+		Use:   "version",
+		Short: "Print the version.",
+		Args:  cobra.NoArgs,
+		Run: func(runner exec.Runner, args []string, flags *flags) error {
+			return runner.Version()
+		},
+	}
+)
+
+// cmdTemplate contains the static parts of a cobra.Command such as
+// documentation that we want to store outside of runtime creation.
+//
+// We do not just store cobra.Commands as in theory they have fields
+// with types such as slices that if we were to return a blind copy,
+// would mean that both the global cmdTemplate and the runtime
+// cobra.Command would point to the same location. By making a new
+// struct, we can also do more fancy templating things like prepending
+// the Short description to the Long description for consistency, and
+// have our own abstractions for the Run command.
+type cmdTemplate struct {
+	// Use is the one-line usage message.
+	// This field is required.
+	Use string
+	// Short is the short description shown in the 'help' output.
+	// This field is required.
+	Short string
+	// Long is the long message shown in the 'help <this-command>' output.
+	// The Short field will be prepended to the Long field with a newline
+	// when applied to a *cobra.Command.
+	// This field is optional.
+	Long string
+	// Expected arguments.
+	// This field is optional.
+	Args cobra.PositionalArgs
+	// Run is the command to run given an exec.Runner, args, and flags.
+	// This field is required.
+	Run func(exec.Runner, []string, *flags) error
+	// BindFlags binds flags to the *pflag.FlagSet on Build.
+	// There is no corollary to this on *cobra.Command.
+	// This field is optional, although usually will be set.
+	// We need to do this before run as the flags are populated
+	// before Run is called.
+	BindFlags func(*pflag.FlagSet, *flags)
+}
+
+// Build builds a *cobra.Command from the cmdTemplate.
+func (c *cmdTemplate) Build(exitCodeAddr *int, stdin io.Reader, stdout io.Writer, stderr io.Writer, flags *flags) *cobra.Command {
+	command := &cobra.Command{}
+	command.Use = c.Use
+	command.Short = c.Short
+	if c.Long != "" {
+		command.Long = fmt.Sprintf("%s\n%s", c.Short, c.Long)
+	}
+	command.Args = c.Args
+	command.Run = func(_ *cobra.Command, args []string) {
+		checkCmd(exitCodeAddr, stdin, stdout, stderr, args, flags, c.Run)
+	}
+	if c.BindFlags != nil {
+		c.BindFlags(command.PersistentFlags(), flags)
+	}
+	return command
+}
```

Comment:
Nice! If you like this more, you could map the direct field translations at the time of the `*cobra.Command` construction:
```
command := &cobra.Command{
  Args:   c.Args,
  Short:  c.Short,
  Use:    c.Use,
  Run:    func(_ *cobra.Command, args []string) { ... },
}
...
```

Reply:
I liked running down the fields in order if that's OK?

Reply:
Yup, fine with me!